app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java

```diff
@@ -764,6 +743,29 @@ public void onFailure(@NonNull Call<Search> call, @NonNull Throwable t) {
 
     }
 
+    public void getProductsByAllergen(final String allergen, final int page, final OnAllergensCallback onAllergensCallback) {
+        apiService.getProductsByAllergen(allergen, page).enqueue(new Callback<Search>() {
+            @Override
+            public void onResponse(Call<Search> call, Response<Search> response) {
+                if (!response.isSuccessful()) {
+                    onAllergensCallback.onAllergensResponse(false, null);
+                    return;
+                }
+
+                if (Integer.valueOf(response.body().getCount()) == 0) {
+                    onAllergensCallback.onAllergensResponse(false, null);
```

Comment:
Check the count (number of results)  in the _ProductBrowsingListActivity_ rather than checking it here. Passing `false` as the method argument makes the method `loadData` defined in `ProductBrowsingListActivity` always display **No internet connection error** in case it's `isResponeOk` argument is false. This can be solved something similar to the case _SearchType.CATEGORY:_ (Line number 403 of `ProductBrowslingListActivity`).